pkg/booleanpolicy/default_policies_test.go

```diff
@@ -3106,6 +3126,82 @@ func (suite *DefaultPoliciesTestSuite) TestLivenessProbePolicyCriteria() {
 	}
 }
 
+func (suite *DefaultPoliciesTestSuite) getViolations(policy *storage.Policy, dep EnhancedDeployment) Violations {
+	matcher, err := BuildDeploymentMatcher(policy)
+	suite.NoError(err, "deployment matcher creation must succeed")
+	violations, err := matcher.MatchDeployment(nil, dep)
+	suite.NoError(err, "deployment matcher run must succeed")
+	suite.Empty(violations.ProcessViolation)
+	return violations
+}
+
+func (suite *DefaultPoliciesTestSuite) TestNetworkPolicyFields() {
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		return
+	}
+
+	testCases := map[string]struct {
+		netpolsApplied augmentedobjs.NetworkPoliciesApplied
```

Comment:
Could you please model this as pointer (`netpolsApplied *augmentedobjs.NetworkPoliciesApplied`) and add a test case with `nil`?